File: sdesheet/src/main/java/com/example/sdesheet/Day04/TwoSum.java
package com.example.sdesheet.Day04;

import java.util.*;

public class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> tracker = new HashMap<Integer, Integer>();
        int length = nums.length;
        for (int i = 0; i < length; i++) {
            if (tracker.containsKey(nums[i])) {
                int key = tracker.get(nums[i]);
                return new int[]{key, i};
            }
            else {
                tracker.put(target - nums[i], i);
            }
        }
        
        return new int[]{};
    }
}

File: src/neetcode/ArraysAndHashing/two_sum.py
from typing import List


class TwoSum:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_map = {}
        for i, n in enumerate(nums):
            if target - n in num_map:
                return [i, num_map[target - n]]
            num_map[n] = i
        return []